proposal/README.md

```diff
@@ -0,0 +1,260 @@
+    - We may consider adding a rolling UTXO set hash for local node storage to make accessing expected UTXO set hashes faster, and may augment the assumeutxo commitment with its value.
+1. (Far down the road) a consensus commitment to the UTXO set hash at a given height may be considered. The snapshot and background validation process may be reused as we migrate to a more compact representation of the UTXO set, e.g. [utreexo](https://www.youtube.com/watch?v=edRun-6ubCc), [UHS](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2018-May/015967.html), or [accumulators](https://eprint.iacr.org/2018/1188).
+
+If parts of that were incomprehensible, keep reading for details.
+
+Right now you can help by reviewing this proposal and the draft code.
+
+#### Resources
+
+- Github issue: https://github.com/bitcoin/bitcoin/issues/15605
+- Phase 1 draft implementation: https://github.com/bitcoin/bitcoin/pull/15606
+
+#### Already familiar?
+
+If you've been following this discussion and already understand the basics of
+`assumeutxo`, you can probably skip down to the [*Security* section](#security).
+
+#### Acknowledgements
+
+I'd like to thank the following people for helping with this proposal, though they should not be held accountable for any dumb mistakes I may have made in the preparation of this document and related code:
+
+Suhas Daftuar, Pieter Wuille, Greg Maxwell, Matt Corallo, Alex Morcos, Dave Harding, AJ Towns, Sjors Provoost, Marco Falke, Russ Yanofsky, and Jim Posen.
+
+
+## Basics
+
+### What's a UTXO snapshot?
+
+It's a serialized version of the unspent transaction output (UTXO) set, as seen from a certain height in the chain. The serialized UTXOs are packaged with some metadata: e.g.
+  - total number of coins contained in the snapshot,
+  - the block header for the latest block encapsulated in a snapshot (its "base"),
+
+and a few other things. You can see the full data structure (in its current form) in the [assumeutxo pull request](https://github.com/bitcoin/bitcoin/pull/15606/commits/f3a5106f62cb519cdb013c0a1a2b9b5461d927fc).
+
+### What's `assumeutxo`?
+
+It's a piece of data embedded in the source code that commits to the hash of a serialized UTXO set which is considered valid for some chain height. The final format of this commitment is still subject to debate because generating it is computationally expensive, and its structure affects how we store and transmit the serialized UTXO set to and from peers. But at the moment it is a simple SHA256-based hash of the UTXO set contents generated by the existing [`GetUTXOStats()` utility](https://github.com/bitcoin/bitcoin/blob/91a25d1e711bfc0617027eee18b9777ff368d6b9/src/rpc/blockchain.cpp#L944-L981).
+
+
+### Okay... what's the use of these things?
+
+We can use the two of them to drastically reduce the amount of time it takes to bootstrap a usable Bitcoin node under a security model that seems acceptable.
```

Comment:
Suggest replacing "the two of them" with "UTXO snapshots and `assumeutxo`" for clarity.

Reply:
Fixed; thanks for the look Schmidty!